mlir/include/mlir/Dialect/Bufferization/IR/BufferizableOpInterface.h

```diff
@@ -351,6 +354,16 @@ struct BufferizationOptions {
   /// used.
   UnknownTypeConverterFn unknownTypeConverterFn = nullptr;
 
+  // Use during type conversion to determine the memory space for memref based
+  // on the originanl tensor type
+  GetMemorySpaceFn getMemorySpaceFn = nullptr;
```

Comment:
I think we don't need both `defaultMemorySpace` and `getMemorySpaceFn`. Can you rename this field to `defaultMemorySpaceFn` and give it a default lambda of `return Attribute();`? Also clarify in the documentation of the field that this lambda is only used when the memory space cannot be inferred.